src/main/webapp/app/shared/markdown-editor/monaco/markdown-editor-monaco.component.html

```diff
@@ -98,6 +105,82 @@
                                 </div>
                             </div>
                         }
+                        <!-- Special case: Lecture attachment reference. -->
+                        @if (displayedActions.lecture) {
+                            <button [matMenuTriggerFor]="subMenuLecture" type="button" class="btn btn-sm py-0 m-0">
+                                <span [jhiTranslate]="displayedActions.lecture.translationKey"></span>
+                                <fa-icon [icon]="faAngleDown" class="ms-1" />
+                            </button>
+                            <mat-menu #subMenuLecture="matMenu" type="button">
+                                @for (lecture of displayedActions.lecture.lecturesWithDetails; track lecture.id) {
+                                    <button class="ps-1 me-1" mat-menu-item [matMenuTriggerFor]="lectureMenuUnits" type="button">
+                                        {{ lecture.title }}
+                                    </button>
+                                    <mat-menu #lectureMenuUnits="matMenu" type="button">
+                                        <button
```

Comment:
I'm not entirely sure if this is possible within a mat-menu, as some mat-components require a specific tag structure. However, you could try adding an ng-template for the buttons.

Reply:
I tried this - unfortunately, it does break the menu. Specifically, the hover trigger to open a sub-menu no longer works. The sub-menu still opens on click, but the original menu closes, so the sub-menu is just floating around on the page.